desc_matcher: {
  target_pcd: "/media/chamo/095d3ecf-bef8-469d-86a3-fe170aec49db/nio/loop1/chamo_out.pcd",
  source_bag: "/media/chamo/095d3ecf-bef8-469d-86a3-fe170aec49db/nio/loop3/chamo_c.bag",
  source_traj: "/media/chamo/095d3ecf-bef8-469d-86a3-fe170aec49db/nio/loop3/traj.txt",
  out_addr: "/media/chamo/095d3ecf-bef8-469d-86a3-fe170aec49db/nio/loop1/no_floor.pcd",
  is_generate_map: false,
  SegMatch: {
    segmentation_radius_m: 50.0,
    segmentation_height_above_m: 999.0,
    segmentation_height_below_m: 999.0,

    filter_boundary_segments: true,
    boundary_radius_m: 999.0,
    filter_duplicate_segments: true,
    centroid_distance_threshold_m: 2,
    min_time_between_segment_for_matches_s: 60,
    check_pose_lies_below_segments: false,

    LocalMap: {
      voxel_size_m: 0.10,
      min_points_per_voxel: 1,
      radius_m: 50,
      min_vertical_distance_m: -999.0,
      max_vertical_distance_m: 999.0,
      neighbors_provider_type: "KdTree",
    },
    
    Segmenters: {
      segmenter_type: "IncrementalEuclideanDistance", # IncrementalSmoothnessConstraints, SimpleSmoothnessConstraints
      min_cluster_size: 100,
      max_cluster_size: 15000,
      radius_for_growing: 0.2
    },

    Descriptors: {
      descriptor_types: ["CNN", "EigenvalueBased"],
      cnn_model_path: "/media/chamo/095d3ecf-bef8-469d-86a3-fe170aec49db/nio/ckpt/",
      semantics_nn_path: "/media/chamo/095d3ecf-bef8-469d-86a3-fe170aec49db/nio/ckpt_semtics/"
    },

    Classifier: {
     threshold_to_accept_match: 0.65,
     n_nearest_neighbours: 25, #25
     knn_feature_dim: 64, #32
     enable_two_stage_retrieval: false,
     apply_hard_threshold_on_feature_distance: false,
     feature_distance_threshold: 100, #25
     
     normalize_eigen_for_knn: false,
     normalize_eigen_for_hard_threshold: false,
     max_eigen_features_values: [2493.5, 186681.0, 188389.0, 0.3304, 188388.0, 1.0899, 0.9987],
     do_not_use_cars: true,
    },

    GeometricConsistency: {
      recognizer_type: "Simple", #"Simple", "Partitioned"
      resolution: 0.6,
      min_cluster_size: 6,
      max_consistency_distance_for_caching: 10.0,
    }
  }
}
